javascript/npm/npm-shrinkwrap-requires/npm-shrinkwrap.json

```diff
@@ -0,0 +1,82 @@
+{
+  "name": "efda-npm-shrinkwrap",
+  "version": "0.0.1",
+  "lockfileVersion": 1,
+  "requires": true,
+  "dependencies": {
+    "balanced-match": {
+      "version": "0.4.2",
+      "resolved": "https://registry.npmjs.org/balanced-match/-/balanced-match-0.4.2.tgz",
+      "integrity": "sha1-yz8+PHMtwPAe5wtAPzAuYddwmDg="
+    },
+    "boom": {
+      "version": "2.10.1",
+      "resolved": "https://registry.npmjs.org/boom/-/boom-2.10.1.tgz",
+      "integrity": "sha1-OciRjO/1eZ+D+UkqhI9iWt0Mdm8=",
+      "requires": {
+        "hoek": "2.x.x"
```

Comment:
Can you generate a module in the requires section such that its version string is something with version numbers such as “2” or “2.1” instead of a version constraint. And also some concrete version number such as “2.1.2”.

Reply:
Many thanks @spencerxiao! I have added `wrappy 1` for the example with incomplete version number, and for concrete version number, we have `concat-map 0.0.1`.